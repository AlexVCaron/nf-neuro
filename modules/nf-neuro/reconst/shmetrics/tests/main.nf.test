nextflow_process {

    name "Test Process RECONST_SHMETRICS"
    script "../main.nf"
    process "RECONST_SHMETRICS"

    tag "modules"
    tag "modules_nfneuro"
    tag "reconst"
    tag "reconst/shmetrics"

    tag "subworkflows"
    tag "subworkflows/load_test_data"

    setup {
            run("LOAD_TEST_DATA", alias: "LOAD_DATA") {
                script "../../../../../subworkflows/nf-neuro/load_test_data/main.nf"
                process {
                    """
                    input[0] = Channel.from( [ "processing.zip" ] )
                    input[1] = "test.load-test-data"
                    """
                }
            }
        }

    test("reconst - shmetrics") {

        config "./nextflow.config"

        when {
            process {
                """
                input[0] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ], // meta map
                        file("\${test_data_directory}/fodf.nii.gz"),
                        [],
                        file("\${test_data_directory}/fa.nii.gz"),
                        file("\${test_data_directory}/md.nii.gz")
                    ]
                }
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    niftiMD5SUM(process.out.peaks.get(0).get(1), 6),
                    niftiMD5SUM(process.out.peak_indices.get(0).get(1), 6),
                    niftiMD5SUM(process.out.peak_values.get(0).get(1), 6),
                    niftiMD5SUM(process.out.afd_max.get(0).get(1), 6),
                    niftiMD5SUM(process.out.afd_total.get(0).get(1), 6),
                    niftiMD5SUM(process.out.afd_sum.get(0).get(1), 6),
                    niftiMD5SUM(process.out.nufo.get(0).get(1), 6),
                    niftiMD5SUM(process.out.vent_mask.get(0).get(1), 6),
                    process.out.versions
                ).match() }
            )
        }
    }

    test("reconst - shmetrics - stub-run") {
        tag "stub"
        options "-stub-run"

        when {
            process {
                """
                input[0] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ], // meta map
                        file("\${test_data_directory}/fodf.nii.gz"),
                        [],
                        file("\${test_data_directory}/fa.nii.gz"),
                        file("\${test_data_directory}/md.nii.gz")
                    ]
                }
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match() }
            )
        }

    }
}
