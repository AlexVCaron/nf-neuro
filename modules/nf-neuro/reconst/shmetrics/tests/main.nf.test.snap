{
    "reconst - shmetrics - stub-run": {
        "content": [
            [
                "versions.yml:md5,9bd2b90c1a1148e83d8037357a93ef04"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-09-22T20:18:36.560384817"
    },
    "reconst - shmetrics": {
        "content": [
            "test__peaks.nii.gz",
            "test__peak_indices.nii.gz",
            "test__peak_values.nii.gz:md5:header,431b1cae3ef9b6fb9fd012a7c4bbbf2f,data,72943c0123dd662b6d237ffe92373b85",
            "test__afd_max.nii.gz:md5:header,c5e41f89848f91c53a9a7be44970d4b1,data,a7cb5485f23516738c2dac6c50db9c61",
            "test__afd_total.nii.gz:md5:header,c5e41f89848f91c53a9a7be44970d4b1,data,7345a2dfed03d48715a505916a8bbfc5",
            "test__afd_sum.nii.gz:md5:header,c5e41f89848f91c53a9a7be44970d4b1,data,0160b88bb94bbd947928a3a0c8736f91",
            "test__nufo.nii.gz:md5:header,c5e41f89848f91c53a9a7be44970d4b1,data,786f036ddab6d83d14d27664dfc5258f",
            "test__ventricles_mask.nii.gz:md5:header,c5e41f89848f91c53a9a7be44970d4b1,data,c99be28705f53b65f3dd406c53e273e6",
            [
                "versions.yml:md5,9bd2b90c1a1148e83d8037357a93ef04"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-09-24T00:51:38.107144086"
    }
}